File: sample/dracula.ts
type VampireProfile = {
  location: string;
  birthDate: number;
  deathDate: number;
  weaknesses: readonly string[];
};

class Vampire {
  constructor(private readonly profile: VampireProfile) {}

  get age(): number {
    return this.profile.deathDate - this.profile.birthDate;
  }
}

const dracula = new Vampire({
  location: "Transylvania",
  birthDate: 1428,
  deathDate: 1476,
  weaknesses: ["Sunlight", "Garlic"],
});

const alucard = new Vampire({
  location: "Wallachia",
  birthDate: 1458,
  deathDate: 1510,
  weaknesses: ["Holy Water"],
});

console.log({ draculaAge: dracula.age, alucardAge: alucard.age });

File: sample/dracula.c
#include <stdlib.h>
#include <stdio.h>

struct Vampire {
    const char *location;
    int birth_date;
    int death_date;
    const char *weaknesses[2];
};

static int get_age(const struct Vampire *vampire) {
    return vampire->death_date - vampire->birth_date;
}

int main() {
    struct Vampire dracula = {
        .location = "Transylvania",
        .birth_date = 1428,
        .death_date = 1476,
        .weaknesses = {"Sunlight", "Garlic"},
    };

    struct Vampire alucard = {
        .location = "Wallachia",
        .birth_date = 1458,
        .death_date = 1510,
        .weaknesses = {"Holy Water", "Silver"},
    };

    printf("Dracula: %d, Alucard: %d\n", get_age(&dracula), get_age(&alucard));
    return EXIT_SUCCESS;
}